crates/solidity/outputs/npm/package/src/ast/generated/frontend.ts

```diff
@@ -0,0 +1,4329 @@
+// This file is generated automatically by infrastructure scripts. Please don't edit by hand.
+
+import { ast_internal } from "../../generated";
+import { RuleNode, TokenNode } from "../../cst";
+import { RuleKind, TokenKind } from "../../kinds";
+
+function once<T>(factory: () => T): () => T {
+  let value: T | null = null;
+  return () => {
+    if (value === null) {
+      value = factory();
+    }
+    return value;
+  };
+}
+
+export class SourceUnit {
+  private readonly fields = once(() => ast_internal.pickSequence(this.cst, RuleKind.SourceUnit));
+
+  public constructor(public readonly cst: RuleNode) {}
+
+  public readonly members: () => SourceUnitMembers | null = once(() => {
+    const field = this.fields()[0];
+    return field ? new SourceUnitMembers(field as RuleNode) : null;
+  });
```

Comment:
I wonder why you are using `once` for both the field picking and the individual fields themselves. Could `fields` not do the construction and the field getters just be simple array accesses i.e. using member access syntax?

Reply:
Sounds good. Can do that!

Reply:
Good point. Artifact of an earlier iteration. Will update it.